agent/dns.go

```diff
@@ -1159,49 +1160,18 @@ func (d *DNSServer) lookupServiceNodes(cfg *dnsConfig, lookup serviceLookup) (st
 		ServiceTags: []string{lookup.Tag},
 		TagFilter:   lookup.Tag != "",
 		QueryOptions: structs.QueryOptions{
-			Token:      d.agent.tokens.UserToken(),
-			AllowStale: cfg.AllowStale,
-			MaxAge:     cfg.CacheMaxAge,
+			Token:            d.agent.tokens.UserToken(),
+			AllowStale:       cfg.AllowStale,
+			MaxAge:           cfg.CacheMaxAge,
+			UseCache:         cfg.UseCache,
```

Comment:
This was just an extra bug being fixed here?

Reply:
Nope, behaviour should not have changed. 

The QueryOptions are different here because previously they were handled below. `cfg.UseCache` was checked directly before calling `cache.Get`. Now `rpcclient/health` does the same, but it looks at the QueryOptions.UseCache, so it is being set here. Same for `MaxStale`.

Reply:
Thanks for the info.